### PR TITLE
PatronDetails: disable force checkout for items on loan

### LINKED ISSUE #81: Disable force checkout when item on loan
In the backoffice, from the patron details page, when scanning for a book that is already on loan, the checkout should fail.
Show a text to explain that the book is already on loan and the current loan should be cancelled first.

### PR DESCRIPTION
**Pull request by KonstantinaStoikou**
Closes #81 

Screenshot from 2020-07-28 14-33-47

### CODE REVIEW

File: src/lib/pages/backoffice/Patron/PatronDetails/ItemsSearch/ItemsResultsList/ItemsResultsList.js [hunk 1]
```diff
@@ -52,56 +52,74 @@ export default class ItemsResultsList extends Component {
...
+              Force Checkout
+            </Button>
+          }
+          open={isModalOpen}
+          onClose={this.toggleModal}
+          closeIcon
+        >
+          <Header
+            icon="exclamation triangle"
+            content="Force Checkout of Missing Item"
```
Comment (topless) [inline src/lib/pages/backoffice/Patron/PatronDetails/ItemsSearch/ItemsResultsList/ItemsResultsList.js:87]: There might be other reasons that the item cannot circulate other than `missing`. Maybe `Force Checkout` is enough.

We should align how we type Check-in and check-out 

- check-in
- checkin
- Checkin
- CheckIn
- Check-In
..

For variable names I have used `CheckOut` everywhere
For display (header, sidebar etc) `Check-out`

what do you suggest @KonstantinaStoikou ? and maybe the naming police @kprzerwa has an opinion.

File: src/lib/pages/backoffice/Patron/PatronDetails/ItemsSearch/ItemsResultsList/ItemsResultsList.js [hunk 2]
```diff
@@ -52,56 +52,74 @@ export default class ItemsResultsList extends Component {
...
+          </Modal.Content>
+          <Modal.Actions>
+            <Button color="red" onClick={this.toggleModal}>
+              <Icon name="remove" /> No
+            </Button>
+            <Button
+              color="green"
+              onClick={() =>
+                this.onClickCheckoutHandler(
+                  item.metadata.document_pid,
```
Comment (topless) [inline src/lib/pages/backoffice/Patron/PatronDetails/ItemsSearch/ItemsResultsList/ItemsResultsList.js:103]: The linter will complain here,
`{ document_pid: documentPid } = item.metadata` and then we use `documentPid`

File: src/lib/pages/backoffice/Patron/PatronDetails/ItemsSearch/ItemsResultsList/ItemsResultsList.js (hunk 1 above)
Comment (KonstantinaStoikou) [reply src/lib/pages/backoffice/Patron/PatronDetails/ItemsSearch/ItemsResultsList/ItemsResultsList.js:87]: I found those things:
- https://grammarist.com/spelling/checkout-check-out/
- https://grammarist.com/spelling/check-in/
that explain the correct naming

File: src/lib/pages/backoffice/Patron/PatronDetails/ItemsSearch/ItemsResultsList/ItemsResultsList.js [hunk 3]
```diff
@@ -52,56 +52,74 @@ export default class ItemsResultsList extends Component {
       circulationStatus
     ) && invenioConfig.ITEMS.canCirculateStatuses.includes(row.metadata.status)
       ? buttonCheckout
-      : this.renderForceCheckoutModal(row);
+      : this.renderForceCheckoutModal(
+          row,
+          circulationStatus === 'ITEM_ON_LOAN'
```
Comment (ntarocco) [inline src/lib/pages/backoffice/Patron/PatronDetails/ItemsSearch/ItemsResultsList/ItemsResultsList.js:57]: what about using circulation.activeStates instead?

File: src/lib/pages/backoffice/Patron/PatronDetails/ItemsSearch/ItemsResultsList/ItemsResultsList.js (hunk 1 above)
Comment (ntarocco) [reply src/lib/pages/backoffice/Patron/PatronDetails/ItemsSearch/ItemsResultsList/ItemsResultsList.js:87]: 👍  to harmonize check-in/out everywhere with the correct spelling!

File: src/lib/pages/backoffice/Patron/PatronDetails/ItemsSearch/ItemsResultsList/ItemsResultsList.js [hunk 4]
```diff
@@ -52,56 +52,74 @@ export default class ItemsResultsList extends Component {
...
+          onClose={this.toggleModal}
+          closeIcon
+        >
+          <Header
+            icon="exclamation triangle"
+            content="Force Checkout of Missing Item"
+          />
+          <Modal.Content>
+            <p>
+              You are creating a loan for a missing item, would you like to
```
Comment (ntarocco) [inline src/lib/pages/backoffice/Patron/PatronDetails/ItemsSearch/ItemsResultsList/ItemsResultsList.js:91]: Agree about the missing, there might be other reasons. Something like:
`You are creating a loan for a physical copy that cannot circulate. Are you sure?`

File: src/lib/pages/backoffice/Patron/PatronDetails/ItemsSearch/ItemsResultsList/ItemsResultsList.js [hunk 5]
```diff
@@ -52,56 +52,74 @@ export default class ItemsResultsList extends Component {
...
+                )
+              }
+            >
+              <Icon name="checkmark" /> Yes
+            </Button>
+          </Modal.Actions>
+        </Modal>
+        {isItemOnLoan && (
+          <Popup
+            content="Item is already on loan. To checkout this item, first cancel the current loan."
```
Comment (ntarocco) [inline src/lib/pages/backoffice/Patron/PatronDetails/ItemsSearch/ItemsResultsList/ItemsResultsList.js:116]:
```suggestion
            content="Physical copy already on loan. To checkout this copy, first cancel the current loan."
```

File: src/lib/pages/backoffice/Patron/PatronDetails/ItemsSearch/ItemsResultsList/ItemsResultsList.js (hunk 2 above)
Comment (KonstantinaStoikou) [reply src/lib/pages/backoffice/Patron/PatronDetails/ItemsSearch/ItemsResultsList/ItemsResultsList.js:103]: The linter is not complaining. `item` is a function argument, it's not a prop, why do I have to destruct it?

**Pull request merged**